Merging Vector Database to Develop

Amaan17 opened the pull request:
Issue:  #29 #31

# Root Cause
The initial backend setup was not optimized for a production environment. The key issues were:

**Excessive Docker Image Size:** The backend image was over 10GB due to the default PyTorch installation including the full CUDA toolkit, which is unnecessary for a CPU-based deployment.

**Inefficient Build Process:** The Dockerfile was downloading and installing dependencies multiple times across different stages, leading to extremely long build times (up to 40 minutes).

**Suboptimal Content Splitting:** The document ingestion process lacked semantic awareness, which could result in lower-quality search results by splitting sentences in half.

# Solution 

**Optimized the Dockerfile:** It now builds much faster and creates a smaller image by installing the CPU-only version of PyTorch.

**Added spaCy:** The ingestion script now uses spaCy to split documents into proper sentences, which improves the quality of our search results.

# How Has This Been Tested? 
<!--- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration --> 

# Checklist: 
- [ ✔️] I have linked issue to this Pull Request 
- [✔️ ] I have performed a self-review of my code 
- [ ✔️] I have commented my code, particularly in hard-to-understand areas 
- [✔️ ] I have made corresponding changes to the documentation 
- [ ] I have added tests that prove my fix is effective or that my feature works 
- [ ] New and existing unit tests pass locally with my changes

Gunnraj-D committed:
Feat/gunnraj d/#1 database and models (#9)

# Task
Establishes SQLite database and models, with methods to interact with
the database in predefined ways.

# Root Cause
We currenlty do not have a database.

# Solution 
2 python classes that represent a project and transcription. These
classes have methods that allow you to interact with the database.

# How Has This Been Tested? 
Rohith made test cases using unittests. First enter bash terminal by
typing "bash" into your terminal in vscode. Then navigate to the server
directory. Then run the command "python -m unittest". This will execute
the tests.

# Checklist: 
- [ ] I have linked issue to this Pull Request 
- [✔] I have performed a self-review of my code 
- [ ✔] I have commented my code, particularly in hard-to-understand
areas
- [ ] I have made corresponding changes to the documentation 
- [ ✔] I have added tests that prove my fix is effective or that my
feature works
- [✔ ] New and existing unit tests pass locally with my changes

---------

Co-authored-by: Gunnraj98 <gdha0007@student.monash.edu>
Co-authored-by: chap989 <rcha0087@student.monash.edu>

YueYaoYan committed:
#14 Create LICENSE (#15)

YueYaoYan committed:
Feat/emily salievski/#3 homepage template (#13)

# Task
Resolves #3 

# Root Cause
Homepage is needed to display our content to client.

# Solution 
- Utilised high fidility prototype and programmed react client
homepage.This includes:
  - AI chat assistant
  - navigation bar
  - analysis steps
  - upload file
  - transcript section
- Added documentation
- Fixed docker npm install issue

# How Has This Been Tested? 
Unit test cannot be generated frot frontend component. Manual test was
done by @EmilySalievski and @YueYaoYan to ensure features are working as
expected.

# Checklist: 
- [x] I have linked issue to this Pull Request 
- [x] I have performed a self-review of my code 
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation 
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] New and existing unit tests pass locally with my changes

---------

Co-authored-by: EmilySalievski <esal0014@student.monash.edu>

YueYaoYan committed:
#23 GitHub action workflow (#25)

# Task
Resolves #23

# Root Cause
CI/CD pipeline has not been setup previously. This is needed to ensure
code quality and build confidence.

# Solution 
Added workflow, this includes:
- Docker compose
- Build confidence test for frontend (ensuring it can build)
- Running pytest for backend

# How Has This Been Tested? 
Tested with personal branch, this PR will also contain check. After
merge, PR check will be turned on ensuring everyone has to pass the test
in order to merge to main or develop.

guiltyeaten2 committed:
Feat/joel/#18 integration: Integration of everything except VD (#32)

# Task
Integrated all feature excepts VD from milestone 2#ISSUE 

# Root Cause
lacking current iteration

# Solution 
Integration

# How Has This Been Tested? 
Done with manual test

# Checklist: 
- [ 1] I have linked issue to this Pull Request 
- [ 1] I have performed a self-review of my code 
- [ 1] I have commented my code, particularly in hard-to-understand
areas
- [ 1] I have made corresponding changes to the documentation 
- [ 1] I have added tests that prove my fix is effective or that my
feature works
- [ 1] New and existing unit tests pass locally with my changes

---------

Co-authored-by: EmilySalievski <esal0014@student.monash.edu>
Co-authored-by: raj0s <aryamanrajpal04@gmail.com>
Co-authored-by: RackyDay4 <rdid0002@student.monash.edu>
Co-authored-by: EamonSom <141641155+EamonSom@users.noreply.github.com>
Co-authored-by: Mohidkz05 <mkha0155@student.monash.edu>
Co-authored-by: lisag678 <lisagorman00@gmail.com>
Co-authored-by: asin0142 <achal1739@gmail.com>
Co-authored-by: Tommy <tkip0001@student.monash.edu>
Co-authored-by: Gunnraj98 <gdha0007@student.monash.edu>

Amaan17 commented:
Thanks Guys!

Mohidkz05 reviewed (approved):
seems to be no conflicts further adequacy will be judged through team 1's integration.

RackyDay committed:
Update README.md to include link to notion backlog (#24)

# Task
Resolves #ISSUE 

# Root Cause
<!--- Please describe why this change is required. What's the motive?
-->

# Solution 
<!--- Please describe your code changes in detail for developers. --> 

# How Has This Been Tested? 
<!--- Please describe the tests that you ran to verify your changes.
Provide instructions so we can reproduce. Please also list any relevant
details for your test configuration -->

# Checklist: 
- [ ] I have linked issue to this Pull Request 
- [ ] I have performed a self-review of my code 
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation 
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] New and existing unit tests pass locally with my changes
